https://github.com/mesonbuild/meson/commit/0a3a9fa0c3ebf45c94d9009a59cead571cbecf7b

This seems to break static linking with our illumos linker.

diff -wpruN '--exclude=*.orig' a~/mesonbuild/linkers/linkers.py a/mesonbuild/linkers/linkers.py
--- a~/mesonbuild/linkers/linkers.py	1970-01-01 00:00:00
+++ a/mesonbuild/linkers/linkers.py	1970-01-01 00:00:00
@@ -208,7 +208,8 @@ class ArLinker(ArLikeLinker):
 
     def get_std_link_args(self, is_thin: bool) -> T.List[str]:
         # FIXME: osx ld rejects this: "file built for unknown-unsupported file format"
-        if is_thin and not mesonlib.is_osx():
+        # FIXME: illumos ld also rejects this: "unknown file type"
+        if is_thin and not mesonlib.is_osx() and not mesonlib.is_sunos():
             return self.std_thin_args
         else:
             return self.std_args
